gateway/middleware.go

```diff
@@ -403,6 +403,20 @@ func (t BaseMiddleware) ApplyPolicies(session *user.SessionState) error {
 							}
 						}
 
+						for _, t := range v.RestrictedTypes {
```

Comment:
do we support partitioning of restricted types?

Reply:
I know it is unrelated to this PR but could we make iteration variables names more meaningful?

like `v -> accessRight` and `k -> apiID`

Reply:
You are right, it is very unclear but better to make it in another PR.